metricbeat/mb/testing/testdata.go

```diff
@@ -150,14 +151,29 @@ func TestDataFiles(t *testing.T, module, metricSet string) {
 }
 
 // TestDataFilesWithConfig run tests for a testdata config
-func TestDataFilesWithConfig(t *testing.T, module, metricSet string, config DataConfig) {
+func TestDataFilesWithConfig(t *testing.T, module, metricSet string, config DataConfig, testPrefix ...string) {
 	t.Helper()
-	ff, err := filepath.Glob(filepath.Join(config.Path, "*."+config.Suffix))
+
+	// the location of the read files
+	location := filepath.Join(config.Path, "*."+config.Suffix)
+
+	// if this function was called from data_test.go then the testPrefix should be 1
+	if len(testPrefix) != 0 {
+		// the prefix for read and write files should be ../../../module/moduleName/metricsetName
+		prefix := filepath.Join(testPrefix[0], module, metricSet)
+		location = filepath.Join(prefix, location)
+
+		// the prefix needs to be appended to the path of the expected files and the original files
+		config.WritePath = filepath.Join(prefix, config.WritePath)
```

Comment:
That really worries me that we override the configuration. Can we avoid this please?

Reply:
Maybe it would be nicer if those are initialised inside the "constructor" that initialised the default config -> https://github.com/elastic/beats/blob/89bcc33a9a90cbacceaa61fceec0e8d23073574c/metricbeat/mb/testing/testdata.go#L112

Reply:
We don't have access to the prefix when that function is called, that is why we need to change it at that moment.

Reply:
Got it, thanks for the extra info!